### PR TITLE
Management command for verifying MOTECH connection SSL certs

### PR DESCRIPTION
**Pull request by joxl**
## Summary

Add new management command (`verify_motech_connection_certs`) for verifying SSL certificates of MOTECH connection URLs.

For aiding detection of SSL trust chain regressions due to #29313.

Jira: [SAAS-12157](https://dimagi-dev.atlassian.net/browse/SAAS-12157)

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/motech/management/commands/verify_motech_connection_certs.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        errors = []
+        failures = []
+        urls = [v for (k, v) in sorted(netlocs.items())]
+        for url in with_progress_bar(urls, oneline=(not verbose)):
+            try:
+                debug(f"HEAD {url}")
+                requests.head(url, verify=(castore or True), timeout=timeout)
+            except SSLError:
+                failures.append(url)
+            except requests.RequestException as exc:
```
Comment (snopoke) [inline corehq/motech/management/commands/verify_motech_connection_certs.py:64]: Won't you also need to catch the other exceptions: `CannotResolveHost, OAuth2Error`

**Review by snopoke — APPROVED** [review]
This seems like it supersedes `ping_motech_connections` if you added an option to test the `http` connections. Maybe rename it and add an `--ssl-only` option to only verify SSL connections?

File: corehq/motech/management/commands/verify_motech_connection_certs.py (hunk 1 above)
Comment (joxl) [reply corehq/motech/management/commands/verify_motech_connection_certs.py:64]: No, `requests.request()` calls won't raise any of those. They originated from custom motech model/util logic initiated by the `get_requests()` and subsequent `send_request_unlogged()` calls.

https://github.com/dimagi/commcare-hq/blob/f086d611ce5955c815544eb6d2a3262c9384897a/corehq/motech/management/commands/ping_motech_connections.py#L20-L22

**Comment by joxl** [comment]
> This seems like it supersedes `ping_motech_connections` if you added an option to test the `http` connections. Maybe rename it and add an --ssl-only option to only verify SSL connections?

I'm ok merging the two together. It would be a complete overhaul of the connection test, which is why I split out a separate "ssl only" tool.  In reality, even the `requests` library is heavy-handed for performing SSL verification (`pyOpenSSL` would likely suffice, eliminating the HTTP overhead), but using `requests` was a simpler implementation with a negligible penalty.

Testing the actual response for each URL is an entirely different can of worms because it requires the added complexity of negotiating all the HTTP specifics (authentication, cannot dedup by domain, etc). Your current script seems to do this well, which is why I left it alone.

I still think that the tools perform similar enough functionality that it makes sense to merge them under one command.  I'll push a new commit to do this.

**Review by millerdev — APPROVED** [review]

**Review by biyeun — APPROVED** [review]

File: corehq/motech/management/commands/ping_motech_connections.py [hunk 2]
```diff
@@ -1,34 +1,204 @@
...
+            help="Write failures to CSV %(metavar)s (default=STDOUT)")
 
     def handle(self, *args, **options):
+        self.verbose = options["verbosity"] > 1
+        timeout = options["connect_timeout"]
+        castore = options["ca_bundle"]
+
+        # sanity-check options
+        if timeout < 0:
+            raise CommandError(f"Invalid timeout value: {timeout}")
```
Comment (snopoke) [inline corehq/motech/management/commands/ping_motech_connections.py:50]: If you want you can use an action to validate the range: https://github.com/dimagi/commcare-hq/blob/354ef5bd94dab77a5bed76c6f46d1502c9e346d0/corehq/util/argparse_types.py#L25

**Review by snopoke — APPROVED** [review]
nit: rename command to `verify_motech_connections`

**Pull request merged**